### PR TITLE
helm: Document missing k8sService kubeConfigPath bpf.mapDynamicSizeRatio

### PR DESCRIPTION
**Pull request by vincentmli**
Document missing k8sService kubeConfigPath bpf.mapDynamicSizeRatio

Signed-off-by: Vincent Li <v.li@f5.com>

Please ensure your pull request adheres to the following guidelines:

- [ ] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [ ] All code is covered by unit and/or runtime tests where feasible.
- [ ] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [ ] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [ ] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [ ] Provide a title or release-note blurb suitable for the release notes.
- [ ] Thanks for contributing!

Fixes: #issue-number

```release-note
Document missing k8sService kubeConfigPath bpf.mapDynamicSizeRatio
```

### CODE REVIEW

File: install/kubernetes/cilium/templates/cilium-configmap.yaml [hunk 1]
```diff
@@ -274,7 +274,7 @@ data:
 
 
 
-{{- if hasKey .Values.bpf "mapDynamicSizeRatio" }}
+{{- if .Values.bpf.mapDynamicSizeRatio }}
   # Specifies the ratio (0.0-1.0) of total system memory to use for dynamic
   # sizing of the TCP CT, non-TCP CT, NAT and policy BPF maps.
   bpf-map-dynamic-size-ratio: {{ .Values.bpf.mapDynamicSizeRatio | quote }}
```
Comment (kaworu) [inline install/kubernetes/cilium/templates/cilium-configmap.yaml]: From the comment I'm guessing that `0.0` is a valid configuration value, but my understanding is that after the patch it would be ignored. Is it a fair statement?

**Review by kaworu — CHANGES_REQUESTED** [review]
Hi @vincentmli and thank you for the PR! One question about `mapDynamicSizeRatio`.

Also small request: could you remove [the `hubble.flowBufferSize` part from the cilium configmap template](https://github.com/cilium/cilium/blob/d5a1a041dc29eb5bf56fa670a90ec293188e9b64/install/kubernetes/cilium/templates/cilium-configmap.yaml#L729-L732) on the way please (also from [the spelling word list](https://github.com/cilium/cilium/blob/d5a1a041dc29eb5bf56fa670a90ec293188e9b64/Documentation/spelling_wordlist.txt#L426)) :pray: ? If I understand correctly we should have done it a while ago :sweat_smile: cc @rolinh

**Comment by rolinh** [comment]
> Also small request: could you remove [the `hubble.flowBufferSize` part from the cilium configmap template](https://github.com/cilium/cilium/blob/d5a1a041dc29eb5bf56fa670a90ec293188e9b64/install/kubernetes/cilium/templates/cilium-configmap.yaml#L729-L732) on the way please (also from [the spelling word list](https://github.com/cilium/cilium/blob/d5a1a041dc29eb5bf56fa670a90ec293188e9b64/Documentation/spelling_wordlist.txt#L426)) pray ? If I understand correctly we should have done it a while ago sweat_smile cc @rolinh

Indeed, it looks like it was overlooked in [this commit](https://github.com/cilium/cilium/commit/2e0427a6a5958485064e217fc4244f8f003c64e2) and it should indeed be removed.

File: install/kubernetes/cilium/templates/cilium-configmap.yaml (hunk 1 above)
Comment (vincentmli) [reply install/kubernetes/cilium/templates/cilium-configmap.yaml]: what does `0.0` ratio do? no system memory for dynamic sizing? I could look into the code to see what `0.0` could do.

Comment (vincentmli) [reply install/kubernetes/cilium/templates/cilium-configmap.yaml]: based on the code, I don't think  `dynamicSizeRatio` with `0.0` is used in the code, doesn't make sense with `0.0`, so we should correct the statement that dynamicSizeRatio should be `0.0 < dynamicSizeRatio <= 1.0`

```
        // Allow the range (0.0, 1.0] because the dynamic size will anyway be
        // clamped to the table limits. Thus, a ratio of e.g. 0.98 will not lead
        // to 98% of the total memory being allocated for BPF maps.
        dynamicSizeRatio := vp.GetFloat64(MapEntriesGlobalDynamicSizeRatioName)
        if 0.0 < dynamicSizeRatio && dynamicSizeRatio <= 1.0 {
                vms, err := mem.VirtualMemory()
                if err != nil || vms == nil {
                        log.WithError(err).Fatal("Failed to get system memory")
                }
                c.calculateDynamicBPFMapSizes(vp, vms.Total, dynamicSizeRatio)
                c.BPFMapsDynamicSizeRatio = dynamicSizeRatio
        } else if dynamicSizeRatio < 0.0 {
                return fmt.Errorf("specified dynamic map size ratio %f must be ≥ 0.0", dynamicSizeRatio)
        } else if dynamicSizeRatio > 1.0 {
                return fmt.Errorf("specified dynamic map size ratio %f must be ≤ 1.0", dynamicSizeRatio)
        }       
```

Comment (vincentmli) [reply install/kubernetes/cilium/templates/cilium-configmap.yaml]: so we have default dynamicSizeRatio set to `0.0` if version < 1.8, but still not used

```
{{- $defaultBpfMapDynamicSizeRatio := 0.0 -}}

{{- /* Default values when 1.8 was initially deployed */ -}}
{{- if semverCompare ">=1.8" (default "1.8" .Values.upgradeCompatibility) -}}

  {{- $defaultBpfMapDynamicSizeRatio = 0.0025 -}}

{{- if .Values.bpf.mapDynamicSizeRatio }}
  # Specifies the ratio (0.0-1.0] of total system memory to use for dynamic
  # sizing of the TCP CT, non-TCP CT, NAT and policy BPF maps.
  bpf-map-dynamic-size-ratio: {{ .Values.bpf.mapDynamicSizeRatio | quote }}
{{- else if ne $defaultBpfMapDynamicSizeRatio 0.0 }}
  # Specifies the ratio (0.0-1.0) of total system memory to use for dynamic
  # sizing of the TCP CT, non-TCP CT, NAT and policy BPF maps.
  bpf-map-dynamic-size-ratio: {{ $defaultBpfMapDynamicSizeRatio | quote }}
{{- end }}

```

**Review by kaworu — APPROVED** [review]
Thanks!

File: install/kubernetes/cilium/templates/cilium-configmap.yaml (hunk 1 above)
Comment (kaworu) [reply install/kubernetes/cilium/templates/cilium-configmap.yaml]: Thanks for investigating!

File: install/kubernetes/cilium/templates/cilium-configmap.yaml [hunk 2]
```diff
@@ -521,16 +521,11 @@ data:
   auto-direct-node-routes: {{ .Values.autoDirectNodeRoutes | quote }}
 
 {{- if .Values.bandwidthManager }}
-{{- if typeIs "bool" .Values.bandwidthManager }}
-  # DEPRECATED: this block should be removed in 1.13
-  enable-bandwidth-manager: {{ .Values.bandwidthManager | quote }}
```
Comment (aanm) [inline install/kubernetes/cilium/templates/cilium-configmap.yaml:517]: It seems this change is unrelated to the PR

File: install/kubernetes/cilium/templates/cilium-configmap.yaml [hunk 3]
```diff
@@ -726,10 +721,6 @@ data:
   # the buffer size is set to the default monitor queue size.
   hubble-event-queue-size: {{ .Values.hubble.eventQueueSize | quote }}
 {{- end }}
-{{- if hasKey .Values.hubble "flowBufferSize" }}
-  # DEPRECATED: this block should be removed in 1.11
-  hubble-flow-buffer-size: {{ .Values.hubble.flowBufferSize | quote }}
-{{- end }}
```
Comment (aanm) [inline install/kubernetes/cilium/templates/cilium-configmap.yaml:747]: same as this one.

File: install/kubernetes/cilium/templates/cilium-configmap.yaml [hunk 4]
```diff
@@ -274,12 +274,12 @@ data:
...
-  # Specifies the ratio (0.0-1.0) of total system memory to use for dynamic
+{{- if .Values.bpf.mapDynamicSizeRatio }}
+  # Specifies the ratio (0.0-1.0] of total system memory to use for dynamic
   # sizing of the TCP CT, non-TCP CT, NAT and policy BPF maps.
   bpf-map-dynamic-size-ratio: {{ .Values.bpf.mapDynamicSizeRatio | quote }}
 {{- else if ne $defaultBpfMapDynamicSizeRatio 0.0 }}
-  # Specifies the ratio (0.0-1.0) of total system memory to use for dynamic
+  # Specifies the ratio (0.0-1.0] of total system memory to use for dynamic
   # sizing of the TCP CT, non-TCP CT, NAT and policy BPF maps.
   bpf-map-dynamic-size-ratio: {{ $defaultBpfMapDynamicSizeRatio | quote }}
```
Comment (aanm) [inline install/kubernetes/cilium/templates/cilium-configmap.yaml:285]: How can detect a cluster that's performing an upgrade for which it never explicitly set `mapDynamicSizeRatio` in its values? My understanding is that if we are setting a default of 0.0025 that means clusters that never set `mapDynamicSizeRatio` will default to 0.0025 which will resize bpf maps.

**Review by aanm — CHANGES_REQUESTED** [review]

File: install/kubernetes/cilium/templates/cilium-configmap.yaml (hunk 2 above)
Comment (vincentmli) [reply install/kubernetes/cilium/templates/cilium-configmap.yaml:517]: this is  clean up I mentioned in the commit message

File: install/kubernetes/cilium/templates/cilium-configmap.yaml (hunk 3 above)
Comment (vincentmli) [reply install/kubernetes/cilium/templates/cilium-configmap.yaml:747]: yep, same clean up mentioned in the commit message

File: install/kubernetes/cilium/templates/cilium-configmap.yaml (hunk 4 above)
Comment (vincentmli) [reply install/kubernetes/cilium/templates/cilium-configmap.yaml:285]: so I did a quick test to install cilium 1.11.0 `cilium install --version=v1.11.0 --kube-proxy-replacement strict --helm-set-string=k8sServiceHost=10.3.72.239,k8sServicePort=6443,ipam.operator.clusterPoolIPv4PodCIDR="10.42.0.0/16"`, you can see I did not specify helm option `bpf.mapDynamicSizeRatio`, but the cilium agent has `level=info msg="  --bpf-map-dynamic-size-ratio='0.0025'" subsys=daemon`, so the default `mapDynamicSizeRatio` is set to 0.0025 since version `>=1.8`, this PR change does not change that behavior, am I missing anything?

File: install/kubernetes/cilium/templates/cilium-configmap.yaml (hunk 3 above)
Comment (vincentmli) [reply install/kubernetes/cilium/templates/cilium-configmap.yaml:747]: to give some background, I am doing piece meal to document missing helm options and helm clean up  starting from  https://github.com/cilium/cilium/pull/21195, the initial thought is to  add the missing options and clean up in one PR split in a few commits, but it turned out the change is too big, can be difficult to review and also result in rebase conflicts, kind of pain the in the neck, so it is suggested to do this by piece meal :)

Comment (kaworu) [reply install/kubernetes/cilium/templates/cilium-configmap.yaml:747]: @aanm to be fair [I asked for this removal on the way](https://github.com/cilium/cilium/pull/21817#pullrequestreview-1149183370) since there was already a cleanup diff in the patch, so this one is on me. If you feel strongly I'm of course ok to have it done in another PR.

File: install/kubernetes/cilium/templates/cilium-configmap.yaml (hunk 4 above)
Comment (kaworu) [reply install/kubernetes/cilium/templates/cilium-configmap.yaml:285]: @aanm note that `$defaultBpfMapDynamicSizeRatio` is untouched by this PR, my understanding is that a >=v1.8 cluster without explicit configuration of `mapDynamicSizeRatio` already default to `0.0025`, see [here](https://github.com/cilium/cilium/blob/b4a4b62296e8fed33c669fd1dd885f7645c0a16d/install/kubernetes/cilium/templates/cilium-configmap.yaml#L21).

In fact, the [CLI default value of `0.0`](https://github.com/cilium/cilium/blob/b4a4b62296e8fed33c669fd1dd885f7645c0a16d/daemon/cmd/daemon_main.go#L815) is not in the accepted range (see [here](https://github.com/cilium/cilium/blob/b4a4b62296e8fed33c669fd1dd885f7645c0a16d/pkg/option/config.go#L3503-L3518)) but that's arguably out of the scope of this PR.

**Review by kaworu — CHANGES_REQUESTED** [review]
@vincentmli looks like we forgot to update the accepted range in [the CLI flag documentation](https://github.com/cilium/cilium/blob/b4a4b62296e8fed33c669fd1dd885f7645c0a16d/daemon/cmd/daemon_main.go#L815) from `(0.0-1.0)` to `(0.0-1.0]`, could you please update it and run `make -C Documentation update-cmdref` :pray:?

**Comment by vincentmli** [comment]
> @vincentmli looks like we forgot to update the accepted range in [the CLI flag documentation](https://github.com/cilium/cilium/blob/b4a4b62296e8fed33c669fd1dd885f7645c0a16d/daemon/cmd/daemon_main.go#L815) from `(0.0-1.0)` to `(0.0-1.0]`, could you please update it and run `make -C Documentation update-cmdref` 🙏?

no problem, I think we should also change the default from `0.0` to `0.0025` at the same line, right?

**Comment by vincentmli** [comment]
> > @vincentmli looks like we forgot to update the accepted range in [the CLI flag documentation](https://github.com/cilium/cilium/blob/b4a4b62296e8fed33c669fd1dd885f7645c0a16d/daemon/cmd/daemon_main.go#L815) from `(0.0-1.0)` to `(0.0-1.0]`, could you please update it and run `make -C Documentation update-cmdref` 🙏?
> 
> no problem, I think we should also change the default from `0.0` to `0.0025` at the same line, right?

hm, https://github.com/cilium/cilium/blob/master/daemon/cmd/daemon_main.go#L815
`	flags.Float64(option.MapEntriesGlobalDynamicSizeRatioName, 0.0, "Ratio (0.0-1.0) of total system memory to use for dynamic sizing of CT, NAT and policy BPF maps. Set to 0.0 to disable dynamic BPF map sizing (default: 0.0)")
	option.BindEnv(Vp, option.MapEntriesGlobalDynamicSizeRatioName)`

based on above, it looks  for  version `<1.8`, `0.0` is a valid input default value to disable map dynamic sizing, since `0.0025` is default now when version `>=1.8`, should we remove the `Set to 0.0 to disable dynamic BPF map sizing (default: 0.0)` also ?

File: install/kubernetes/cilium/templates/cilium-configmap.yaml (hunk 4 above)
Comment (aanm) [reply install/kubernetes/cilium/templates/cilium-configmap.yaml:284]: @kaworu @vincentmli could we compromise and leave the changes on this PR but making them in a different commit, what do you think?

File: install/kubernetes/cilium/templates/cilium-configmap.yaml (hunk 3 above)
Comment (aanm) [reply install/kubernetes/cilium/templates/cilium-configmap.yaml:747]: @kaworu @vincentmli could we compromise and leave the changes on this PR but making them in a different commit, what do you think?

File: install/kubernetes/cilium/templates/cilium-configmap.yaml (hunk 4 above)
Comment (aanm) [reply install/kubernetes/cilium/templates/cilium-configmap.yaml:285]: Thank you for taking a look. LGTM

**Review by aanm — APPROVED** [review]

**Review by qmonnet — APPROVED** [review]
Agreed with André that the clean-up part should be in a dedicated commit. Looks good to me otherwise, thanks!

File: install/kubernetes/cilium/templates/cilium-configmap.yaml (hunk 3 above)
Comment (vincentmli) [reply install/kubernetes/cilium/templates/cilium-configmap.yaml:747]: fine by me :)

**Review by kaworu — APPROVED** [review]
@vincentmli thanks!

**Review by aanm — CHANGES_REQUESTED** [review]
@vincentmli it looks the CI failures are legitimate

**Comment by vincentmli** [comment]
> @vincentmli it looks the CI failures are legitimate

yep
```
Stderr:

	 Error: template: cilium/templates/cilium-configmap.yaml:508:14: executing "cilium/templates/cilium-configmap.yaml" at <.Values.bandwidthManager.enabled>: can't evaluate field enabled in type interface {}
```
not helm expert, I wonder if following would fix the problem, 
```
diff --git a/install/kubernetes/cilium/templates/cilium-configmap.yaml b/install/kubernetes/cilium/templates/cilium-configmap.yaml
index 240e174e9c..83a67a7403 100644
--- a/install/kubernetes/cilium/templates/cilium-configmap.yaml
+++ b/install/kubernetes/cilium/templates/cilium-configmap.yaml
@@ -504,7 +504,7 @@ data:
 
   auto-direct-node-routes: {{ .Values.autoDirectNodeRoutes | quote }}
 
-{{- if .Values.bandwidthManager }}
+{{- if haskey .Values "bandwidthManager" }}
 {{- if .Values.bandwidthManager.enabled }}
   enable-bandwidth-manager: {{ .Values.bandwidthManager.enabled | quote }}
   enable-bbr: {{ .Values.bandwidthManager.bbr | quote }}
```

**Comment by vincentmli** [comment]
@aanm I made a change that could fix the issue, could you trigger `/test` for me? Thanks!

**Comment by vincentmli** [comment]
still same error

> /test
> 
> Job 'Cilium-PR-K8s-1.23-kernel-5.4' failed:
> 
> Click to show.
> If it is a flake and a GitHub issue doesn't already exist to track it, comment `/mlh new-flake Cilium-PR-K8s-1.23-kernel-5.4` so I can create one.

still same error, I will look into it

```
 Error: template: cilium/templates/cilium-configmap.yaml:508:14: executing "cilium/templates/cilium-configmap.yaml" at <.Values.bandwidthManager.enabled>: can't evaluate field enabled in type interface {}
```

**Comment by vincentmli** [comment]
I can generate same error locally after looking at the CI test example that has `--set bandwidthManager=false `

[root@cilium-dev cilium]# helm template  install/kubernetes/cilium --namespace=kube-system  --set bandwidthManager=false  > test.yaml
Error: template: cilium/templates/cilium-configmap.yaml:508:14: executing "cilium/templates/cilium-configmap.yaml" at <.Values.bandwidthManager.enabled>: can't evaluate field enabled in type interface {}

 the problem is `bandwidthManager` bool is removed by this PR, but the CI test still keep using  `--set bandwidthManager=false`, maybe the CI should remove `--set bandwidthManager=false ` ? or any other suggestion ? @aanm

**Comment by aanm** [comment]
>  the problem is `bandwidthManager` bool is removed by this PR, but the CI test still keep using  `--set bandwidthManager=false`, maybe the CI should remove `--set bandwidthManager=false ` ?

@vincentmli yes, I think that's a good approach.

**Comment by vincentmli** [comment]
> > the problem is `bandwidthManager` bool is removed by this PR, but the CI test still keep using  `--set bandwidthManager=false`, maybe the CI should remove `--set bandwidthManager=false ` ?
> 
> @vincentmli yes, I think that's a good approach.

I assume you or  someone will remove the CI `--set bandwidthManager=false ` then :)  I could file CI issue report if necessary

**Comment by aanm** [comment]
> > > the problem is `bandwidthManager` bool is removed by this PR, but the CI test still keep using  `--set bandwidthManager=false`, maybe the CI should remove `--set bandwidthManager=false ` ?
> > 
> > 
> > @vincentmli yes, I think that's a good approach.
> 
> I assume you or someone will remove the CI `--set bandwidthManager=false ` then :) I could file CI issue report if necessary

@vincentmli why can't it be done in this PR? :thinking:

**Comment by vincentmli** [comment]
> @vincentmli why can't it be done in this PR? 🤔

@aanm  I thought CI test code change is only done by cilium committers where the CI test code sit in somewhere in github, not in cilium repo, no? I sure can do it if I can, could you give me direction where to change the CI test code ?

**Comment by aanm** [comment]
@vincentmli I think this patch will do it:
```diff
diff --git a/test/helpers/kubectl.go b/test/helpers/kubectl.go
index 6a512a119e..ceecf8a973 100644
--- a/test/helpers/kubectl.go
+++ b/test/helpers/kubectl.go
@@ -2497,7 +2497,7 @@ func (kub *Kubectl) overwriteHelmOptions(options map[string]string) error {
                addIfNotOverwritten(options, "sessionAffinity", "false")
        }
        if DoesNotRunOnNetNextKernel() {
-               addIfNotOverwritten(options, "bandwidthManager", "false")
+               addIfNotOverwritten(options, "bandwidthManager.enabled", "false")
        }
 
        if RunsWithHostFirewall() {
```

**Comment by vincentmli** [comment]
> @vincentmli I think this patch will do it:
> 
> ```diff
> diff --git a/test/helpers/kubectl.go b/test/helpers/kubectl.go
> index 6a512a119e..ceecf8a973 100644
> --- a/test/helpers/kubectl.go
> +++ b/test/helpers/kubectl.go
> @@ -2497,7 +2497,7 @@ func (kub *Kubectl) overwriteHelmOptions(options map[string]string) error {
>                 addIfNotOverwritten(options, "sessionAffinity", "false")
>         }
>         if DoesNotRunOnNetNextKernel() {
> -               addIfNotOverwritten(options, "bandwidthManager", "false")
> +               addIfNotOverwritten(options, "bandwidthManager.enabled", "false")
>         }
>  
>         if RunsWithHostFirewall() {
> ```

Great, I just grep search string bandwidthManager and found above code, thinking about how to change it :). should this be in the same commit of removing bandwidthManager?

**Comment by aanm** [comment]
> > @vincentmli I think this patch will do it:
> > ```diff
> > diff --git a/test/helpers/kubectl.go b/test/helpers/kubectl.go
> > index 6a512a119e..ceecf8a973 100644
> > --- a/test/helpers/kubectl.go
> > +++ b/test/helpers/kubectl.go
> > @@ -2497,7 +2497,7 @@ func (kub *Kubectl) overwriteHelmOptions(options map[string]string) error {
> >                 addIfNotOverwritten(options, "sessionAffinity", "false")
> >         }
> >         if DoesNotRunOnNetNextKernel() {
> > -               addIfNotOverwritten(options, "bandwidthManager", "false")
> > +               addIfNotOverwritten(options, "bandwidthManager.enabled", "false")
> >         }
> >  
> >         if RunsWithHostFirewall() {
> > ```
> 
> Great, I just grep search string bandwidthManager and found above code, thinking about how to change it :). should this be in the same commit of removing bandwidthManager?

Yes, I think it's better to have both changes in the same commit.

**Comment by vincentmli** [comment]
> Yes, I think it's better to have both changes in the same commit.

great, I added  ` Co-authored-by: André Martins <andre@cilium.io>` and pushed the changes, hope it will work this time :)

**Comment by vincentmli** [comment]
the failed test agent log, I suspect since we removed `bandwidthManager` bool, which is conflicting with `helpers.CiliumStableHelmChartVersion`  that still has `bandwidthManager` bool, I am not familiar with the test code, @aanm any more idea :)


2022-10-31T12:43:30.609177973Z level=fatal msg="Incorrect config-map flag enable-bandwidth-manager" error="strconv.ParseBool: parsing \"map[enabled:false]\": invalid syntax" subsys=config



      It("Tests upgrade and downgrade from a Cilium stable image to master", func() {
                var assertUpgradeSuccessful func()
                assertUpgradeSuccessful, cleanupCallback =
                        InstallAndValidateCiliumUpgrades(
                                kubectl,
                                helpers.CiliumStableHelmChartVersion,
                                helpers.CiliumStableVersion,
                                helpers.CiliumLatestHelmChartVersion,
                                helpers.GetLatestImageVersion(),
                        )
                assertUpgradeSuccessful()
        })

**Comment by vincentmli** [comment]
I think probably we should just remove          ` addIfNotOverwritten(options, "bandwidthManager.enabled", "false")` completely, what is the idea to set bandwidthManager to false in test, if bandwidthManager is not required in the test, why set it and set it false

**Comment by aanm** [comment]
@vincentmli we need to change all of these "bandwidthManager" with "bandwidthManager.enabled" https://github.com/cilium/cilium/blob/e2f0dbda6a756b1db23a7bf082fcd34fcf99f61c/test/k8s/bandwidth.go#L116-L143

**Comment by vincentmli** [comment]
> @vincentmli we need to change all of these "bandwidthManager" with "bandwidthManager.enabled"

@aanm ok, I understand above is for fail test
```
[Test Result](https://jenkins.cilium.io/job/Cilium-PR-K8s-1.25-kernel-net-next/593/testReport/) (3 failures / -106)
[Suite-k8s-1.25.K8sDatapathBandwidthTest Checks Bandwidth Rate-Limiting Checks Pod to Pod bandwidth, vxlan tunneling](https://jenkins.cilium.io/job/Cilium-PR-K8s-1.25-kernel-net-next/593/testReport/junit/Suite-k8s-1/25/K8sDatapathBandwidthTest_Checks_Bandwidth_Rate_Limiting_Checks_Pod_to_Pod_bandwidth__vxlan_tunneling/)
[Suite-k8s-1.25.K8sDatapathBandwidthTest Checks Bandwidth Rate-Limiting Checks Pod to Pod bandwidth, geneve tunneling](https://jenkins.cilium.io/job/Cilium-PR-K8s-1.25-kernel-net-next/593/testReport/junit/Suite-k8s-1/25/K8sDatapathBandwidthTest_Checks_Bandwidth_Rate_Limiting_Checks_Pod_to_Pod_bandwidth__geneve_tunneling/)
[Suite-k8s-1.25.K8sDatapathBandwidthTest Checks Bandwidth Rate-Limiting Checks Pod to Pod bandwidth, direct routing](https://jenkins.cilium.io/job/Cilium-PR-K8s-1.25-kernel-net-next/593/testReport/junit/Suite-k8s-1/25/K8sDatapathBandwidthTest_Checks_Bandwidth_Rate_Limiting_Checks_Pod_to_Pod_bandwidth__direct_routing/)
```

what about fail test `Tests upgrade and downgrade from a Cilium stable image to master` ?

**Comment by aanm** [comment]
@vincentmli something similar as [this](https://github.com/cilium/cilium/commit/ad1e0c4876d4e48539abfccb18807b9d0d010239#diff-5c1af29398c5b8b90b92ed032a7e93bbdd900f933641c270946df9225abb7143L407-L412) will work. (If the direct link doesn't work it's the file test/k8sT/Updates.go lines 407-412

**Comment by vincentmli** [comment]
> @vincentmli something similar as [this](https://github.com/cilium/cilium/commit/ad1e0c4876d4e48539abfccb18807b9d0d010239#diff-5c1af29398c5b8b90b92ed032a7e93bbdd900f933641c270946df9225abb7143L407-L412) will work. (If the direct link doesn't work it's the file test/k8sT/Updates.go lines 407-412

sorry I am not following how above would resolve the problem, line 407 - 412 seems irrelevant here by removing version check and set `"agent":               "false ",`. 

here is the standard out error from the test:

```
Standard Error
12:52:12 STEP: Running BeforeAll block for EntireTestsuite K8sUpdates
12:52:12 STEP: Ensuring the namespace kube-system exists
12:52:12 STEP: WaitforPods(namespace="kube-system", filter="-l k8s-app=cilium-test-logs")
12:52:13 STEP: WaitforPods(namespace="kube-system", filter="-l k8s-app=cilium-test-logs") => <nil>
12:52:13 STEP: Waiting for pods to be terminated
12:52:29 STEP: Deleting Cilium and CoreDNS
12:52:29 STEP: Waiting for pods to be terminated
12:52:29 STEP: Cleaning Cilium state (28d819eb71af886ef71c92a7fdfaa7215ed99a10)
12:52:30 STEP: Cleaning up Cilium components
12:52:30 STEP: Waiting for Cilium to become ready
12:52:48 STEP: Cleaning Cilium state (v1.11)
12:52:48 STEP: Cleaning up Cilium components
12:52:54 STEP: Waiting for Cilium to become ready
12:53:11 STEP: Deploying Cilium 1.11
12:53:11 STEP: Waiting for Cilium to become ready
FAIL: Timed out after 240.001s.
Timeout while waiting for Cilium to become ready
Expected
    <*errors.errorString | 0xc0008186b0>: {
        s: "only 0 of 2 desired pods are ready",
    }
to be nil
```
it looks to me the test was trying to deploy Cilium stable release version 1.11, but failed, the standard error output order matches `testfunc` order in `func InstallAndValidateCiliumUpgrades()`

```
211         testfunc := func() {
212                 By("Deleting Cilium and CoreDNS")
213                 // Making sure that we deleted the  cilium ds. No assert
214                 // message because maybe is not present
215                 if res := kubectl.DeleteResource("ds", fmt.Sprintf("-n %s cilium", helpers.CiliumNamespace)); !res.WasSuccessful() {
216                         log.Warningf("Unable to delete Cilium DaemonSet: %s", res.OutputPrettyPrint())
217                 }
218 
219                 By("Waiting for pods to be terminated")
220                 ExpectAllPodsTerminated(kubectl)
221 
222                 EventuallyWithOffset(1, func() *helpers.CmdRes {
223                         return kubectl.HelmAddCiliumRepo()
224                 }, time.Second*30, time.Second*1).Should(helpers.CMDSuccess(), "Unable to install helm repository")
225 
226                 // New version must come first given prior CI tests may have run on new Cilium version.
227                 By("Cleaning Cilium state (%s)", newImageVersion)
228                 cleanupCiliumState(filepath.Join(kubectl.BasePath(), helpers.HelmTemplate), newHelmChartVersion, "", newImageVersion, "")
229 
230                 By("Cleaning Cilium state (%s)", oldImageVersion)
231                 cleanupCiliumState(stableChartPath, oldHelmChartVersion, "quay.io/cilium/cilium-ci", oldImageVersion, "")
232 
233                 By("Deploying Cilium %s", oldHelmChartVersion) <===Here deploying cilium 1.1
234 
235                 opts := map[string]string{
236                         "image.tag":                              oldImageVersion,
237                         "operator.image.tag":                     oldImageVersion,
238                         "hubble.relay.image.tag":                 oldImageVersion,
239                         "clustermesh.apiserver.image.tag":        oldImageVersion,
240                         "image.repository":                       "quay.io/cilium/cilium-ci",
241                         "operator.image.repository":              "quay.io/cilium/operator",
242                         "hubble.relay.image.repository":          "quay.io/cilium/hubble-relay-ci",
243                         "clustermesh.apiserver.image.repository": "quay.io/cilium/clustermesh-apiserver-ci",
244                 }
245 
246                 // Eventually allows multiple return values, and performs the assertion
247                 // on the first return value, and expects that all other return values
248                 // are zero values (nil, etc.).
249                 EventuallyWithOffset(1, func() (*helpers.CmdRes, error) {
250                         return kubectl.RunHelm(
251                                 "install",
252                                 stableChartPath,
253                                 "cilium",
254                                 oldHelmChartVersion,
255                                 helpers.CiliumNamespace,
256                                 opts)
257                 }, time.Second*30, time.Second*1).Should(helpers.CMDSuccess(), fmt.Sprintf("Cilium %q was not able to be deployed", oldHe    lmChartVersion))
258 
```

what confused me is that if the test was deploying cilium 1.11 and failed, but why the cilium agent log in the report is showing

```
2022-10-31T12:43:30.609177973Z level=fatal msg="Incorrect config-map flag enable-bandwidth-manager" error="strconv.ParseBool: parsing \"map[enabled:false]\": invalid syntax" subsys=config
```

my assumption is that the test was deploying cilium 1.11, but using the PR helm template with `bandwidthManager.enabled=false` ? that may explain the agent error, if so why not use cilium 1.11 helm template with `bandwidthManager=false`, @aanm Am I missing anything ? :)

**Comment by aanm** [comment]
@vincentmli what I meant was something like this:
```
		hasNewHelmValues := versioncheck.MustCompile(">=1.12.90")
		if hasNewHelmValues(versioncheck.MustVersion(newHelmChartVersion)) {
			opts["bandwidthManager.enabled"] = "false "
		} else {
			opts["bandwidthManager"] = "false "
		}
```

on line 245 of the snippet that you mentioned in the last comment.

**Comment by vincentmli** [comment]
sorry to misunderstand your suggestion :).  I will try the suggested code change.

**Comment by vincentmli** [comment]
@aanm I pushed the new changes, could you invoke test?

**Comment by vincentmli** [comment]
CI failure test seems to related to https://github.com/cilium/cilium/issues/21735, irrelevant to this PR

**Comment by vincentmli** [comment]
@aanm ping :)
